### DISCUSSION TITLE
Multiple users in htpasswd, some read-only?

### DISCUSSION THREAD

**Discussion by schildbach** [distribution/distribution#4316] [Q&A]
I'd like to allow pushing and deleting only for the limited set of machines/user accounts that actually build and push docker images. My servers, the regular consumers of my docker images, shall only be able to read.

Currently I'm using basic auth and sharing an htpasswd user between all clients of the registry. I'd like to add one (or more) htpasswd users that can only read. Is that possible in an easy way?

**Comment by milosgajdos** [comment]
> I'd like to add one (or more) htpasswd users that can only read. Is that possible in an easy way?

This is not possible. I'd recommend looking into implementing a token server following https://distribution.github.io/distribution/spec/auth/token/

Or grab an existing implementation from somewhere.
